src/event/event_handler.rs

```diff
@@ -0,0 +1,87 @@
+//! Helper to run a callback in the libuv main thread.
+
+use std::os::raw::c_void;
+
+use types::*;
+use handle::{Handle, Managed};
+use neon_runtime;
+use neon_runtime::raw;
+use std::sync::Arc;
+use context::Context;
+use context::internal::ContextInternal;
+
+type EventContext<'a> = crate::context::TaskContext<'a>;
+
+struct EventHandlerInner(*mut c_void);
+
+unsafe impl Send for EventHandlerInner {}
+unsafe impl Sync for EventHandlerInner {}
+
+impl Drop for EventHandlerInner {
+    fn drop(&mut self) {
+        unsafe {
+            neon_runtime::handler::delete(self.0);
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct EventHandler(Arc<EventHandlerInner>);
+
+impl EventHandler {
+    #[cfg(feature = "napi-1")]
+    pub fn new<'a, C: Context<'a>, T: Value>(cx: &C, this: Handle<T>, callback: Handle<JsFunction>) -> Self {
+        unimplemented!()
```

Comment:
Maybe include a panic message that explains that this experimental API is not supported in the N-API backend, with a suggestion for an alternative.